vms/proposervm/vm.go

```diff
@@ -353,17 +386,46 @@ func (vm *VM) SetPreference(ctx context.Context, preferred ids.ID) error {
 	// custom VMs. Until the P-chain is modified to target a specific block
 	// time, ProposerMinBlockDelay can be configured in the subnet config.
 	minDelay = math.Max(minDelay, vm.MinBlkDelay)
+	nextStartTime := parentTimestamp.Add(minDelay)
 
-	preferredTime := blk.Timestamp()
-	nextStartTime := preferredTime.Add(minDelay)
-	vm.Scheduler.SetBuildBlockTime(nextStartTime)
+	return nextStartTime, nil
+}
 
-	vm.ctx.Log.Debug("set preference",
-		zap.Stringer("blkID", blk.ID()),
-		zap.Time("blockTimestamp", preferredTime),
-		zap.Time("nextStartTime", nextStartTime),
+func (vm *VM) getPostDurangoSlotTime(
+	ctx context.Context,
+	blkHeight uint64,
+	pChainHeight uint64,
+	parentTimestamp time.Time,
+	currentTime time.Time,
+) (time.Time, error) {
+	delay, err := vm.Windower.MinDelayForProposer(
+		ctx,
+		blkHeight,
+		pChainHeight,
+		parentTimestamp,
+		vm.ctx.NodeID,
+		currentTime,
 	)
-	return nil
+	if errors.Is(err, proposer.ErrNoSlotsScheduledInNextFuture) {
```

Comment:
If we can handle lookahead shorter than our actual assigned slot, I'd recommend a smaller default window (maybe 10 minutes instead of 60).